test/arrayops.jl

```diff
@@ -594,6 +594,33 @@ end
         @test findprev(b, T(1)) isa keytype(b)
         @test findprev(b, T(2)) isa keytype(b)
     end
+
+    @testset "issue #36938" begin
+        # 👎 denotes tests that are broken.
+        @test findfirst(fill(true, (0, 2))) === nothing     # 👎
```

Comment:
Shouldn't these use `@test_broken` with the correct answer then?

Reply:
Looks like all lines with :-1: are currently throwing a BoundsError so this PR is failing CI

Reply:
I've lost track of what needs to happen here. I think my goal was to write tests that should pass, but I don't remember if the current (inconsistent) behavior can be fixed before Julia 2.0, e.g.:

```julia
julia> findfirst(fill(true, (0,)))

julia> findfirst(fill(true, (1, 0)))

julia> findfirst(fill(true, (0, 1)))

julia> findfirst(fill(true, (2, 0)))

julia> findfirst(fill(true, (0, 2)))
ERROR: BoundsError: attempt to access 0×2 Matrix{Bool} at index [1, 1]
```

Obviously failing CI does not help. 

I've long wanted a macro like the one below, then I could write these as `@test_broken`:

```
julia> macro error_as_value(ex)
           quote
               try
                   return Some($(ex))
               catch e
                   return e
               end
           end
       end
@error_as_value (macro with 1 method)

julia> @error_as_value sqrt(4)
Some(2.0)

julia> @error_as_value sqrt(-4)
DomainError(-4.0, "sqrt will only return a complex result if called with a complex argument. Try sqrt(Complex(x)).")
```